codegen/gradle.properties

```diff
@@ -1,2 +1,2 @@
-smithyVersion=1.26.0
+smithyVersion=[1.27.0,1.28.0[
```

Comment:
is this valid gradle syntax? with the hanging open bracket?

Reply:
discussed offline, despite the latter way being more readable/explicit IMO, the former way written in the PR is more gradle-idiomatic. and i agree for most cases its better to follow convention. so im approving.

Reply:
Why a range in the first place? Why not a specific version here? We control the smithy version and when we upgrade, I'm not sure I'm understanding why we are declaring a range of versions to be compatible with.

Reply:
Updated to `1.27.2`